src/app/shared/button.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { EnvironmentProviderService } from '../providers/environment-provider.service';
+import { GlobalAlertService } from './global-alert.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ButtonService {
+  constructor(
+    private environmentService: EnvironmentProviderService,
+    private alert: GlobalAlertService
+  ) {}
+
+  async handleTransactionButton(clickedCallback: () => void) {
+    let eth: any = window.ethereum;
+    if (eth) {
+      let chainId = await eth.request({
```

Comment:
Why not use the `getChainId()` function from `WalletProviderService` here?

Reply:
I was having circular dependency error in one the services. But this works in the button service. I have made the changes and pushed the code